File: src/util/prime/list/BigPrimeList.java
package util.prime.list;

import java.io.*;
import java.nio.file.Files;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

import net.lingala.zip4j.ZipFile;
import org.apache.commons.lang3.StringUtils;

public class BigPrimeList {

    private static BigPrimeList bigPrimeListSingleton = null;
    private List<Integer> list;

    public List<Integer> getList(){
        return list;
    }

    public static BigPrimeList getInstance(){
        if(bigPrimeListSingleton == null){
            bigPrimeListSingleton =  new BigPrimeList();
        }
        return  bigPrimeListSingleton;
    }

    private BigPrimeList(){
        load();
    }

    private void load(){
        System.out.println("Start Loading.....");

        try{

            File tempDir = Files.createTempDirectory("tempDir").toFile();
            String resourcePath = BigPrimeList.class.getResource("data/master.zip.001").getFile();

            try(ZipFile zipFile = new ZipFile(resourcePath)){

                zipFile.extractAll(tempDir.getPath());

                Arrays.stream(tempDir.listFiles()).filter(e -> e.getName().contains("master")).findFirst().ifPresent(masterFile -> {
                    list = readFromFile(masterFile);
                    masterFile.deleteOnExit();
                });

                tempDir.deleteOnExit();
            }

        }catch (IOException e){
            System.out.println(e);
        }

        System.out.println("Done with loading :-)");
    }

    private List<Integer> readFromFile(File masterFile) {

        List<Integer> list = new ArrayList<>();

        try(BufferedReader bufferedReader = new BufferedReader(new FileReader(masterFile))){
            while (bufferedReader.ready()){
                list.addAll(Arrays.stream(bufferedReader.readLine().split(";")).filter(StringUtils::isNotBlank).map(Integer::valueOf).collect(Collectors.toList()));
            }
        }catch(IOException e){
            System.out.println(e);
        }

        return list;
    }
}

File: src/util/prime/list/BigPrimeList.kt
package util.prime.list

import net.lingala.zip4j.ZipFile
import java.io.File

class BigPrimeList {

   var list:List<Int> = listOf()

   private fun load(){
      println("Start Loading.....")
      val tempDir = kotlin.io.path.createTempDirectory("tempDir").toFile()
      val firstZipFile = File(javaClass.getResource("data/master.zip.001").file)
      val zipFile = ZipFile(firstZipFile)
      zipFile.extractAll(tempDir.path)
      val masterFile = tempDir.listFiles().first { e -> e.name.contains("master") }
      list = masterFile.readLines().map { line -> line.split(";").filter { element -> element.isNotEmpty() }.map { element -> element.toInt() } }.flatten().toList()
      masterFile.delete()
      tempDir.delete()
      println("Done with loading :-)")
   }

   fun isPrim(number:Int):Boolean{

      if(list.isEmpty()){
         load()
      }

      if(number > list.last()){
         throw IllegalArgumentException("Param to big :(")
      }

      return list.binarySearch(number) >= 0
   }
}